components/scanners/trivy/vendor/github.com/smithy-security/pkg/sarif/sarif.go

```diff
@@ -356,23 +360,27 @@ func (s *SarifTransformer) mapAffectedPackage(fixes []sarif.Fix, purl packageurl
 	return affectedPackage
 }
 
-func (s *SarifTransformer) detectPackageFromPhysicalLocation(physicalLocation sarif.PhysicalLocation) *packageurl.PackageURL {
+func (s *SarifTransformer) detectPackageFromPhysicalLocation(physicalLocation sarif.PhysicalLocation, ecosyststem string) *packageurl.PackageURL {
 	if s.isSnykURI(*physicalLocation.ArtifactLocation.Uri) {
 		return nil
 	}
 	if p, e := packageurl.FromString(*physicalLocation.ArtifactLocation.Uri); e == nil {
 		return &p
 	}
 	// if we get hinted that this scan is for dependencies and the deps have a specific ecosystem we can assume more things
-	if s.findingsEcosystem != "" {
+	if s.findingsEcosystem != "" || ecosyststem != "" {
+		pkgManager := s.findingsEcosystem
```

Comment:
⛏️ you could simplify with

```go
pkgManager := s.findingsEcosystem
if ecosyststem != "" {
	pkgManager = ecosyststem
}

if pkgManager != "" { ... }
```

Reply:
this is coming from the library update, I've outlined the only hot fixed patch above